mindmeld/auto_annotator.py

```diff
@@ -50,39 +58,60 @@ class Annotator(ABC):
+        language=ENGLISH_LANGUAGE_CODE,
+        locale=ENGLISH_US_LOCALE,
+        overwrite=False,
+        unannotate_supported_entities_only=True,
+        unannotation_rules=None,
+        **kwargs,
+    ):
+        """Initializes an annotator.
 
         Args:
-            app_path (str): The location of the MindMeld app
-            config (dict, optional): A config object to use. This will
-                override the config specified by the app's config.py file.
+            app_path (str): The location of the MindMeld app.
+            annotation_rules (list): List of Annotation rules.
+            language (str, optional): Language as specified using a 639-1/2 code.
+            locale (str, optional): The locale representing the ISO 639-1 language code and \
+                ISO3166 alpha 2 country code separated by an underscore character.
+            overwrite (bool): Whether to overwrite existing annotations with conflicting spans.
+            unannotate_supported_entities_only (bool): Only allow removal of supported entities.
+            unannotation_rules (list): List of Annotation rules.
         """
         self.app_path = app_path
-        self.config = config or get_auto_annotator_config(app_path=app_path)
+        self.language = language
+        self.locale = locale
+        self.overwrite = overwrite
+        self.annotation_rules = annotation_rules or []
+        self.unannotate_supported_entities_only = unannotate_supported_entities_only
+        self.unannotation_rules = unannotation_rules or []
         self._resource_loader = ResourceLoader.create_resource_loader(app_path)
+        self.duckling = DucklingRecognizer.get_instance()
 
-    def _get_file_entities_map(self, action: AnnotatorAction, config):
-        """ Creates a dictionary that maps file paths to entities given
+    def _get_file_entities_map(self, action: AnnotatorAction):
+        """Creates a dictionary that maps file paths to entities given
         regex rules defined in the config.
 
         Args:
             action (AnnotatorAction): Can be "annotate" or "unannotate". Used as a key
                 to access a list of regex rules in the config dictionary.
-            config (dict): Config to use instead of the class config.
 
         Returns:
             file_entities_map (dict): A dictionary that maps file paths in an
                 App to a list of entities.
         """
-        config = config or self.config
         all_file_paths = self._resource_loader.get_all_file_paths()
         file_entities_map = {path: [] for path in all_file_paths}
 
-        if action == AnnotatorAction.ANNOTATE:
-            rules = config[AnnotatorAction.ANNOTATE.value]
-        elif action == AnnotatorAction.UNANNOTATE:
-            rules = config[AnnotatorAction.UNANNOTATE.value]
+        rules = (
+            self.annotation_rules
+            if action == AnnotatorAction.ANNOTATE
+            else self.unannotation_rules
+        )
```

Comment:
I think this is dangerous since if the caller has not explicitly set `ANNOTATE `, the unannotate code will be executed. I think we should just throw an error here if its not ANNOTATE or UNANNOTATE

Reply:
Done